tests/e2e/test_accessibility.py

```diff
@@ -0,0 +1,128 @@
+        '/s20/sample/gradeable/open_homework/grading/details?view=all',
+        '/s20/sample/gradeable/open_homework',
+        '/s20/sample/gradeable/open_team_homework/team',
+        '/s20/sample/gradeable/grades_released_homework_autota',
+        '/s20/sample/notifications',
+        '/s20/sample/notifications/settings',
+        '/s20/sample/gradeable',
+        '/s20/sample/config',
+        '/s20/sample/theme',
+        '/s20/sample/office_hours_queue',
+        '/s20/sample/course_materials',
+        '/s20/sample/forum',
+        '/s20/sample/forum/threads/new',
+        '/s20/sample/forum/categories',
+        '/s20/sample/forum/stats',
+        '/s20/sample/users',
+        '/s20/sample/graders',
+        '/s20/sample/sections',
+        '/s20/sample/student_photos',
+        '/s20/sample/late_days',
+        '/s20/sample/extensions',
+        '/s20/sample/grade_override',
+        '/s20/sample/plagiarism',
+        '/s20/sample/plagiarism/configuration/new',
+        '/s20/sample/reports',
+        '/s20/sample/late_table'
+    ]
+
+    def test_w3_validator(self):
+        # Uncomment this to generate a new baseline for all pages on the website
+        # Then run 'python3 -m unittest e2e.test_accessibility' from inside the tests folder
+        # genBaseline(self)
+
+        # Uncomment this to generate a new baseline for a specific url
+        # url = '' # your url here
+        # genBaseline(self, url)
+
+
+        validatePages(self)
+
+
+
+def validatePages(self):
+    self.log_out()
+    self.log_in(user_id='instructor')
+    self.click_class('sample')
+    with open('/'.join(__file__.split('/')[:-1])+'/accessibility_baseline.json') as f:
+        baseline = json.load(f)
+
+    foundError = False
+    for url in self.urls:
+        self.get(url=url)
+
+        payload = self.driver.page_source
+        headers = {
+          'Content-Type': 'text/html; charset=utf-8'
+        }
+        response = requests.request("POST", "https://validator.w3.org/nu/?out=json", headers=headers, data = payload.encode('utf-8'))
+
+
+
+        for error in response.json()['messages']:
+            # For some reason the test fails to detect this even though when you actually look at the rendered
+            # pages this error is not there. So therefore the test is set to just ignore this error.
+            if error['message'].startswith("Start tag seen without seeing a doctype first"):
+                continue
+
+            if error['message'] not in baseline[url]:
+                error['url'] = url
+                print(json.dumps(error, indent=4, sort_keys=True))
+                foundError = True
+
+    self.assertEqual(foundError, False)
+
+
+
+def genBaseline(self, new_url=None):
+    self.log_out()
+    self.log_in(user_id='instructor')
+    self.click_class('sample')
+
+    baseline = {}
+    urls = self.urls
+    if new_url:
+        with open('/'.join(__file__.split('/')[:-1])+'/accessibility_baseline.json') as f:
```

Comment:
Should we be using `os.path.dirname(os.path.realpath(__file__))` instead? This will always ensure the full path is used rather than the relative.